web/src/tracking/activation.tsx

```diff
@@ -0,0 +1,153 @@
+                          totalCount
+                      }
+                      viewerSettings {
+                          final
+                      }
+                      users {
+                          totalCount
+                      }
+                      currentUser {
+                          usageStatistics {
+                              searchQueries
+                              findRefsActions
+                          }
+                      }
+                  }
+              `
+            : gql`
+                  query {
+                      currentUser {
+                          usageStatistics {
+                              searchQueries
+                              findRefsActions
+                          }
+                      }
+                  }
+              `
+    ).pipe(
+        map(dataOrThrowErrors),
+        map(data => {
+            const authProviders = window.context.authProviders
+            const completed: { [key: string]: boolean } = {
+                didSearch: !!data.currentUser && data.currentUser.usageStatistics.searchQueries > 0,
+                'action:findReferences': !!data.currentUser && data.currentUser.usageStatistics.findRefsActions > 0,
+            }
+            if (isSiteAdmin) {
+                completed.connectedCodeHost = data.externalServices && data.externalServices.totalCount > 0
+                completed.enabledRepository =
+                    data.repositories && data.repositories.totalCount !== null && data.repositories.totalCount > 0
+                if (authProviders) {
+                    completed.enabledSignOn =
+                        data.users.totalCount > 1 || authProviders.filter(p => !p.isBuiltin).length > 0
+                }
+            }
+            return completed
+        })
+    )
+
+/**
+ * Returns a new ActivationStatus instance to use in the web app.
+ *
+ * @param isSiteAdmin determines if site-admin-only activation steps are included.
+ */
+export const newActivationStatus = (isSiteAdmin: boolean) => {
+    const s = new ActivationStatus(
+        [
+            {
+                id: 'connectedCodeHost',
+                title: 'Connect your code host',
+                detail: 'Configure Sourcegraph to talk to your code host and fetch a list of your repositories.',
+                action: (h: H.History) => h.push('/site-admin/external-services'),
+                siteAdminOnly: true,
+            },
+            {
+                id: 'enabledRepository',
+                title: 'Enable repositories',
+                detail: 'Select which repositories Sourcegraph should pull and index from your code host(s).',
+                action: (h: H.History) => h.push('/site-admin/repositories'),
+                siteAdminOnly: true,
+            },
+            {
+                id: 'didSearch',
+                title: 'Search your code',
+                detail: 'Issue a search query over your code.',
```

Comment:
```suggestion
                detail: 'Perform a search query on your code.',
```

Recommend not using "issue" as a verb; see **Avoid ambiguous verbs** at https://sourcegraph.com/github.com/sourcegraph/about@master/-/blob/STYLEGUIDE.md#clarity.